raft-tech/TANF-app
# Issue 967: Dynamically generate DJANGO_SECRET_KEY for initial deployments

### Summary of Changes
Closes #967

Introduces dynamic generation of DJANGO_SECRET_KEY for initial deployments to Cloud.gov and provides a generated default in the event it is not set.

### How to Test
* Remove `DJANGO_SECRET_KEY` from CircleCI environment variables (done on Raft side)
* Run Circle CI checks and ensure no failures result from removing `DJANGO_SECRET_KEY`
* Pick a development app to target
* Delete that development app from Cloud.gov, using the UI or CLI:
```
cf delete [target-app]
```
* Assign the relevant label to deploy this PR to the target app
* Ensure that DJANGO_SECRET_KEY is set to a random string 67 characters in length after the deployment is completed
```
cf env [target-app]
python -c "print(len('[COPY GENERATED KEY HERE]'))"
```
* Reset the Login.gov keypair for the target app
```
./scripts/copy-login-gov-keypair.sh [any-other-dev-app] [target-app]
```
* Finally, log into the deployed app and make sure everything works as expected

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] `DJANGO_SECRET_KEY` is automatically generated for initial deployments to Cloud.gov

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

N/A, configuration change

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

N/A, configuration change

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

N/A, configuration change

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

N/A, configuration change

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*
